dht/dht.go

```diff
@@ -208,8 +238,8 @@ func (nid *nodeID) ByteString() string {
 	return string(buf[:])
 }
 
-type node struct {
-	addr          dHTAddr
+type Node struct {
```

Comment:
I'm not sure there's any value to exposing this at all. The OnQuery user should have no business or interest in real-time updates on the node internals. Nothing is exposed on it. I think it's sufficient just to pass out the node's "source" addr when queries come in, the ID is provided in the msg. Certainly I don't think the external user should be maintaining references to node, as they are dropped and replaced from the node table beyond the control of the user.